cnf-certification-test/accesscontrol/suite.go

```diff
@@ -17,12 +17,294 @@
 
-	logrus.Debugf("%s not moved yet to new framework", common.AccessControlTestKey)
+	logrus.Debugf("Entering %s suite", common.AccessControlTestKey)
+	var env provider.TestEnvironment
+	ginkgo.BeforeEach(func() {
+		provider.BuildTestEnvironment()
+		env = provider.GetTestEnvironment()
+	})
+	// Security Context: non-compliant capabilities
+	testID := identifiers.XformToGinkgoItIdentifier(identifiers.TestSecConCapabilitiesIdentifier)
+	ginkgo.It(testID, ginkgo.Label(testID), func() {
+		TestSecConCapabilities(&env)
+	})
+	// container security context: non-root user
+	testID = identifiers.XformToGinkgoItIdentifier(identifiers.TestSecConNonRootUserIdentifier)
+	ginkgo.It(testID, ginkgo.Label(testID), func() {
+		TestSecConRootUser(&env)
+	})
+	// container security context: privileged escalation
+	testID = identifiers.XformToGinkgoItIdentifier(identifiers.TestSecConPrivilegeEscalation)
+	ginkgo.It(testID, ginkgo.Label(testID), func() {
+		TestSecConPrivilegeEscalation(&env)
+	})
+	// container security context: host port
+	testID = identifiers.XformToGinkgoItIdentifier(identifiers.TestContainerHostPort)
+	ginkgo.It(testID, ginkgo.Label(testID), func() {
+		TestContainerHostPort(&env)
+	})
+	// container security context: host network
+	testID = identifiers.XformToGinkgoItIdentifier(identifiers.TestPodHostNetwork)
+	ginkgo.It(testID, ginkgo.Label(testID), func() {
+		TestPodHostNetwork(&env)
+	})
+	// pod host path
+	testID = identifiers.XformToGinkgoItIdentifier(identifiers.TestPodHostPath)
+	ginkgo.It(testID, ginkgo.Label(testID), func() {
+		TestPodHostPath(&env)
+	})
+	// pod host ipc
+	testID = identifiers.XformToGinkgoItIdentifier(identifiers.TestPodHostIPC)
+	ginkgo.It(testID, ginkgo.Label(testID), func() {
+		TestPodHostIPC(&env)
+	})
+	// pod host pid
+	testID = identifiers.XformToGinkgoItIdentifier(identifiers.TestPodHostPID)
+	ginkgo.It(testID, ginkgo.Label(testID), func() {
+		TestPodHostPID(&env)
+	})
 })
+
+// TestSecConCapabilities verrifies that non compliant capabilities are not present
+func TestSecConCapabilities(env *provider.TestEnvironment) {
+	var badContainers []string
+	var errContainers []string
+	if len(env.Containers) == 0 {
+		ginkgo.Skip("No containers to perform test, skipping")
+	}
+	for _, cut := range env.Containers {
+		if cut == nil {
+			errContainers = append(errContainers, cut.Podname+"."+cut.Data.Name)
+			continue
+		}
+		if cut.Data.SecurityContext != nil && cut.Data.SecurityContext.Capabilities != nil {
+			for _, ncc := range nonCompliantCapabilites {
+				if strings.Contains(cut.Data.SecurityContext.Capabilities.String(), ncc) {
+					tnf.ClaimFilePrintf("Non compliant %s capability detected in container %s. All container caps: %s", ncc, cut.Data.Name, cut.Data.SecurityContext.Capabilities.String())
+					badContainers = append(badContainers, cut.Podname+"."+cut.Data.Name)
+				}
+			}
+			logrus.Infof("test %s", cut.Data.SecurityContext.Capabilities.String())
```

Comment:
I think this trace could be shown before the for loop. Also, what about a more descriptive text like "Testing the existence of %s security context capabiltiy in CNF containers."

Reply:
This was left by mistake, removed it